interface-impl-v2/src/main/java/org/mule/apikit/implv2/v10/model/ParameterImpl.java

```diff
@@ -193,6 +194,14 @@ public Optional<FileProperties> getFileProperties() {
     return empty();
   }
 
+  @Override
+  public boolean isNullable() {
+    if (typeDeclaration instanceof UnionTypeDeclaration) {
+      return hasNilShape((UnionTypeDeclaration) typeDeclaration);
+    }
+    return typeDeclaration instanceof NullTypeDeclaration;
```

Comment:
Does this look better? 🤷 
```suggestion
    return typeDeclaration instanceof NullTypeDeclaration
        || typeDeclaration instanceof UnionTypeDeclaration && hasNilShape((UnionTypeDeclaration) typeDeclaration)
```